services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -134,6 +134,12 @@ public Response createUser(final UserRepresentation rep) {
         if (session.users().getUserByUsername(realm, username) != null) {
             return ErrorResponse.exists("User exists with same username");
         }
+
+        // Check if username is already used as email by other user
+        if (session.users().getUserByEmail(realm, username) != null){
```

Comment:
This doesn't guarantee there isn't another user with this email address as the user could be registered just after this check, but prior to the new user being created.

Reply:
I second that. I'm in favour of closing this PR and looking for a proper solution in the original issue.

Reply:
Aye, agree.
As it's one of my first PRs in OSS project, there is a question I need to ask. 
Should I actually create a PR draft before creating a real one?
Thanks!